api/api-pipeline-selection-v1/src/main/java/com/thoughtworks/go/apiv1/pipelineselection/representers/PipelineSelectionsRepresenter.java

```diff
@@ -33,10 +33,15 @@ public static void toJSON(OutputWriter writer, PipelineSelectionResponse pipelin
             .add("blacklist", pipelineSelectionResponse.getSelectedPipelines().isBlacklist())
             .addChild("pipelines", pipelineGroupsWriter -> {
                 pipelineSelectionResponse.getPipelineConfigs().forEach(pipelineConfigs -> {
-                        List<String> pipelineNames = pipelineConfigs.getPipelines().stream().map(PipelineConfig::getName).map(CaseInsensitiveString::toString).collect(Collectors.toList());
+                    List<String> pipelineNames = pipelineConfigs
+                            .getPipelines().stream()
+                            .map(PipelineConfig::getName)
+                            .map(CaseInsensitiveString::toString)
```

Comment:
can we iter once over the list instead of twice?

`map(pipeline -> pipeline.getName().toString())` ?

Reply:
It's a stream, so it's not actually iterating twice :)